_src/pattern-library/sass/components/_forms.scss

```diff
@@ -0,0 +1,425 @@
+            font-size: font-size(small);
+            color: $helper-text;
+        }
+    }
+
+    &.form-group {
+
+        .field-wrapper { // Generally used for radios and checkboxes
+            margin-bottom: spacing-vertical(small);
+        }
+
+        .helper-text { // Generally used for radios and checkboxes
+            margin-top: spacing-vertical(xx-small);
+            margin-left: 24px;
+        }
+    }
+}
+
+.form-label {
+    display: block;
+    margin-bottom: spacing-vertical(x-small);
+    font-size: font-size(mid-large);
+    color: $label-color;
+
+    &.inline-label {
+        margin-bottom: 0;
+    }
+
+    &.is-active,
+    &.is-selected {
+        color: $label-color-active;
+    }
+}
+
+.helper-text {
+    margin-top: spacing-vertical(x-small);
+}
+
+.required-field:after {
+    display: inline-block;
+    margin-left: ($baseline/4) + px;
+    content: "*";
+    font-size: font-size(small);
+    color: $required-indicator-color;
+}
+
+.input,
+.select,
+.textarea {
+    @include transition(all timing(xx-fast) ease-in-out 0s);
+    display: inline-block;
+    min-width: ($baseline*20) + px; // for the purposes of the demo... can probably get moved elsewhere...
+    padding: ($baseline/2) + px;
+    border: 1px solid $field-default-border;
+    background: $field-default-background;
+    font-family: $font-family-sans-serif;
+    font-size: font-size(base);
+    color: $field-default-text-color;
+
+    &:focus {
+        outline: none;
+    }
+}
+
+.progress {
+    @include transition(all timing(xx-fast) ease-in-out 0s);
+}
+
+
+::-webkit-input-placeholder,
+::-moz-placeholder,
+:-moz-placeholder,
+:-ms-input-placeholder {
+   color: $placeholder-text-color;
+}
+
+// ----------------------------
+// #INPUT TEXT
+// ----------------------------
+.input-text {
+
+    &:focus,
+    &.is-active {
+        border-color: $field-default-border-focus;
+        color: $field-default-text-color-hover;
+    }
+
+    &:disabled,
+    &.is-disabled {
+        border-color: $field-disabled-border;
+        background: $field-disabled-background;
+    }
+
+    &.input-simple {
+        padding: ($baseline/2) + px 0;
+        border-width: 0 0 2px 0;
+
+        &:focus,
+        &.is-active {
+            border-color: $field-simple-border-focus;
+            background: $field-simple-background-focus;
+        }
+    }
+}
+
+// ----------------------------
+// #SELECT
+// ----------------------------
+.wrapper-replace-select {
+    position: relative;
+}
+
+.input-select {
+    height: ($baseline*2) + px;
+    line-height: ($baseline*2) + px;
+
+    &:focus,
+    &.is-active {
+        border-color: $field-default-border-focus;
+        color: $field-default-text-color-hover;
+    }
+
+    &:disabled,
+    &.is-disabled {
+        border-color: $field-disabled-border;
+        background: $field-disabled-background;
+    }
+
+    &.is-replaced {
+        position: absolute;
+        z-index: z-index(mid-front);
```

Comment:
Can we go any lower in depth than this? I'm sensitive about us going too high in the z-index too early.